armi/reactor/converters/uniformMesh.py

```diff
@@ -100,14 +105,26 @@ def _checkConversion(self):
 
     @staticmethod
     def initNewReactor(sourceReactor):
-        """Build an empty version of the new reactor"""
-        # XXX: this deepcopy is extremely wasteful because the assemblies copied
-        # are immediately removed. It's just laziness of getting the same class
-        # of reactor set up.
-        newReactor = copy.deepcopy(sourceReactor)
-        newReactor.core.removeAllAssemblies()
-        newReactor.core.regenAssemblyLists()
-        return newReactor
+        """Build a new, yet empty, reactor with the same settings as sourceReactor
+
+        Parameters
+        ----------
+        sourceReactor : :py:class:`Reactor <armi.reactor.reactors.Reactor>` object.
+            original reactor to be copied
+        """
+        # create new operator
+        newOper = operators.factory(sourceReactor.o.cs)
+        # create new reactor from blueprints used to create sourceReactor
+        bp = blueprints.loadFromCs(sourceReactor.o.cs)
+        newReactor = reactors.Reactor(sourceReactor.o.cs.caseTitle, bp)
+        # create core based on loaded blueprints and do not load the assemblies
+        coreDesign = bp.systemDesigns["core"]
+        coreDesign.construct(sourceReactor.o.cs, bp, newReactor, loadAssems=False)
+        # initialize the interfaces and set some values
+        newOper.initializeInterfaces(newReactor)
+        newOper.r.core.lib = isotxs.readBinary(ISOAA_PATH)
+        newOper.r.core.p.keff = 1.0
```

Comment:
Let's grab the k-effective and library from the old reactor. 

```
newOper.r.core.lib = sourceReactor.core.lib
newOper.r.core.p.keff = sourceReactor.core.p.keff
```

Reply:
I think that k-effective can be removed in the mapping since that is handled later on.